### PR TITLE
Respect pytest directory or file cmd-line parameters when running with --picked

### PR DESCRIPTION
**Pull request by pschroeder89**
**Bug:** 
When running pytest with `pytest ./directory` or `pytest ./directory/file.py` arguments , pytest-picked ignores the directory / file path and runs all edited tests within the whole repository. 

**Fix:**
Use the pytest config object to determine which files / directories were passed as arguments to pytest, and pass those to git in either Unstaged or Branch mode commands.

**Test notes:**
Made edits to the following two test cases:

tests/scenarios/functions/where/test_where.py
tests/api/test_connectors.py

Ran the following:
```
pytest --picked --mode=branch # Collected both test files
pytest tests/api --picked --mode=branch # Collected only the tests/api/test_connectors.py file
pytest tests/scenarios --picked --mode=branch # Collected only the tests/scenarios/functions/where/test_where.py file
```

### CODE REVIEW

File: pytest_picked/plugin.py [hunk 1]
```diff
@@ -41,10 +41,15 @@ def _get_affected_paths(config):
     picked_mode = config.getoption("picked_mode")
     parent_branch = config.getoption("parent_branch")
     test_file_convention = config._getini("python_files")  # pylint: disable=W0212
+    root_dir = str(config.rootdir) + "/"
```
Comment (pschroeder89) [inline pytest_picked/plugin.py:44]: @anapaulagomes let me know if you prefer an "f" string here, instead.

Comment (anapaulagomes) [reply pytest_picked/plugin.py:44]: I do :)

File: .github/workflows/publish_to_pypi.yml [hunk 2]
```diff
@@ -21,4 +21,4 @@ jobs:
       uses: pypa/gh-action-pypi-publish@master
       with:
         user: __token__
-        password: ${{ secrets.PYPI_TOKEN }}
+        password: ${{ secrets.PYPI_TOKEN }}
```
Comment (anapaulagomes) [inline .github/workflows/publish_to_pypi.yml:24]: Could you add an extra line here please?

File: .gitignore [hunk 3]
```diff
@@ -72,4 +73,7 @@ target/
 .python-version
 
 # pytest
-.pytest_cache/
\ No newline at end of file
+.pytest_cache/
+
+# IntelliJ
+.idea
```
Comment (anapaulagomes) [inline .gitignore:79]: Tip: having a global gitignore can avoid having to add this to new repos.

**Review by anapaulagomes — COMMENTED** [review]
I still need to test it locally. Thanks for contributing!

File: .gitignore (hunk 3 above)
Comment (pschroeder89) [reply .gitignore:79]: True, want me to remove this?

Comment (anapaulagomes) [reply .gitignore:79]: No need :) you can leave it there

File: pytest_picked/plugin.py (hunk 1 above)
Comment (pschroeder89) [reply pytest_picked/plugin.py]: Fixed

File: .github/workflows/publish_to_pypi.yml (hunk 2 above)
Comment (pschroeder89) [reply .github/workflows/publish_to_pypi.yml]: Done

**Review by anapaulagomes — COMMENTED** [review]
Hello @pschroeder89. First of all, I wanted to check something with you: shouldn't the options come before the folders? Look at pytest usage: `pytest [options] [file_or_dir] [file_or_dir]`. Running pytest-picked according to pytest usage and using your branch I got this result (which is a bug, indeed):

```
➜  pytest-picked git:(run_against_specific_dir) pytest --picked tests/test_modes.py
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: argument --picked: invalid choice: 'tests/test_modes.py' (choose from 'only', 'first')
```

Before getting into the code I want to touch base about this. I'll try to respond as fast as possible this time. Sorry about this back and forth. By now you can just disregard my comments until we sort this out.

**Comment by pschroeder89** [comment]
> Hello @pschroeder89. First of all, I wanted to check something with you: shouldn't the options come before the folders? Look at pytest usage: `pytest [options] [file_or_dir] [file_or_dir]`. Running pytest-picked according to pytest usage and using your branch I got this result (which is a bug, indeed):
> 
> ```
> ➜  pytest-picked git:(run_against_specific_dir) pytest --picked tests/test_modes.py
> ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
> pytest: error: argument --picked: invalid choice: 'tests/test_modes.py' (choose from 'only', 'first')
> ```
> 
> Before getting into the code I want to touch base about this. I'll try to respond as fast as possible this time. Sorry about this back and forth. By now you can just disregard my comments until we sort this out.

I think the issue here is that the --picked option accepts a choice of only and first. Since one is not being specified, pytest is trying to use the file path as the argument.
In other words, if --picked is used before the file/dir in the pytest command, a choice must be passed to it, as well. This works:
`pytest --picked only ./tests/end2end_with_splunk/test_metrics.py`

**Comment by anapaulagomes** [comment]
I was expecting to have a default option if none is passed so this `pytest --picked tests/` would work. I think we shouldn't change the expected way of passing files or dirs. @pschroeder89

**Comment by pschroeder89** [comment]
Looking at the documentation for argparse (which is used by pytest's `add_option`), optional arguments should go after the filename.
https://docs.python.org/3/howto/argparse.html#introducing-optional-arguments

In pytest's documentation, optional arguments are used interchangeably before and after the filename (take a look at --collect-only in the examples here):
https://docs.pytest.org/en/stable/example/parametrize.html?highlight=options#different-options-for-test-ids

To make the behavior consistent and unambiguous, I could add a new command-line option called `dir` that accepts multiple paths:
```
parser.add_argument('dir', nargs='?', default=os.getcwd())
```
Which could be used like
`pytest --picked --dir tests/abc.py tests/api/`

What are your thoughts @anapaulagomes ?

**Comment by anapaulagomes** [comment]
Sounds good to me @pschroeder89! 🚀

**Pull request closed without merging**